HMS-5287: Add playwright and github action to backend (#956)
content-services/content-sources-backend: README.md

### Start dependency containers

```sh
make compose-up
```

### Import Public Repos

```sh
make repos-import
```

> ### For local development, if you want less Redhat repos try"

Reviewer: ```suggestion
### For local development, if you want less Red Hat repos try"
```

```sh
OPTIONS_REPOSITORY_IMPORT_FILTER=small make repos-import
```

### Run the server!

```sh
make run
```

###

Hit the API:

```sh
curl -H "$( ./scripts/header.sh 9999 1111 )" http://localhost:8000/api/content-sources/v1.0/repositories/
```

### Stop dependency containers

When its time to shut down the running containers:

```sh
make compose-down
```

And clean the volume that it uses by (this stops the container before doing it if it were running):

```sh
make compose-clean
```

> There are other make rules that could be helpful, run `make help` to list them. Some are highlighted below

## Playwright testing

- Ensure that the backend server is running

> - Make sure you are currently on the correct [node version](_playwright-tests/.nvmrc), you can do this easily with nvm.

Reviewer: ```suggestion
- Ensure the correct [node version](_playwright-tests/.nvmrc), is installed an in use:
```

  `cd _playwright-tests` and `nvm use`

> - You also need to copy the [env](_playwright-tests/example.env) file and create your own file at: \_playwright-tests/.env

Reviewer: ```suggestion
- Copy the [env](_playwright-tests/example.env) file and create your own file at: \_playwright-tests/.env
```

Reviewer: Would be nice to say more about that file's contents, perhaps at least "For local testing, only BASE_URL is required"

Author: yeah...

### Import Public Repos

```sh
make repos-import
```

### For local development, if you want less Red Hat repos try"

```sh
OPTIONS_REPOSITORY_IMPORT_FILTER=small make repos-import
```

### Run the server!

```sh
make run
```

###

Hit the API:

```sh
curl -H "$( ./scripts/header.sh 9999 1111 )" http://localhost:8000/api/content-sources/v1.0/repositories/
```

### Stop dependency containers

When its time to shut down the running containers:

```sh
make compose-down
```

And clean the volume that it uses by (this stops the container before doing it if it were running):

```sh
make compose-clean
```

> There are other make rules that could be helpful, run `make help` to list them. Some are highlighted below

## Playwright testing

- Ensure that the backend server is running
- Ensure the correct [node version](_playwright-tests/.nvmrc), is installed and in use:  
  `cd _playwright-tests` and `nvm use`
- Copy the [env](_playwright-tests/example.env) file and create your own file at: \_playwright-tests/.env
  For local development only the BASE_URL is required, which is already set in the example config.

```sh
make playwright
```

OR

```sh
cd _playwright-tests \
&& yarn install \

> && yarn playwright test

Reviewer 2: `yarn playwright install` is in the _playwright-tests directory's README but not here. i missed that step and ran into an error that told me exactly what to do :) but could be nice to have it in both places, and maybe included in the make command?

Reviewer 2: ah and including this in both places makes sense only if it's working for both mac and linux (reading through the slack thread now 😺)
